glean-core/android/src/main/java/mozilla/telemetry/glean/upload/Upload.kt

```diff
@@ -0,0 +1,105 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.telemetry.glean.upload
+
+import org.json.JSONObject
+import com.sun.jna.Structure
+import com.sun.jna.Pointer
+import com.sun.jna.Union
+import mozilla.telemetry.glean.Glean
+import mozilla.telemetry.glean.net.HeadersList
+import mozilla.telemetry.glean.rust.getRustString
+
+@Suppress("ClassNaming")
+@Structure.FieldOrder("tag", "uuid", "path", "body", "headers")
+internal class Upload_Body(
+    // NOTE: We need to provide defaults here, so that JNA can create this object.
+    @JvmField val tag: Byte? = null,
```

Comment:
In both `Upload_Body` and `FfiPingUploadTask` I don't really have control over the naming of properties. The way Rust represents an enum with data over the FFI is with a tagged union, that's what "tag" is here. I can't change the names because that messes up the type mapping. 

Because these raw types I get from the FFI are a bit confusing I created some convience ones (`PingRequest` and `PingUploadTask`) that resemble more closely the types I had in Rust, are easier to consume from Kotlin and make more sense overall. When I call the upload functions from Kotlin I immeditelly convert what I get to a `PingUploadTask`, so that I don't ever really need to deal with these confusing ffi types.

Reply:
All right, I see now. It might be possible to instruct cbindgen to give better names, but it's probably not a good use of time. Please just add a comment above this static that names are generated by cbindgen.

Reply:
As above: It doesn't _have_ to match the generated C header file.
(I for one do like the `tag` though as it's common naming for unions/enums with data)

Reply:
@badboy is correct, I can change the names. My bad.

I also do like `tag`.

Reply:
I don't think `tag` tells a lot about what it contains, so I'm not really a fan of this. No need to over iterate on it, though. Please do still add a comment explaining what it is that it contains.